### ISSUE TITLE
[BUG]NET ERROR: tensors used as indices must be long, byte or bool tensors

### ISSUE THREAD

**Issue by sun1000yao** [MouseLand/cellpose#1079]
**NET ERROR: tensors used as indices must be long, byte or bool tensors**

Hi, When I was using cellpose Zstack to run cyto3 with GPU, this error popped up. Same issue appeared when uncheck the 'use GPU'. Thanks in advance for suggestions and helps.  

**Run log**
2024-12-19 23:37:03,114 [INFO] ** TORCH CUDA version installed and working. **
2024-12-19 23:37:03,114 [INFO] >>>> using GPU (CUDA)
2024-12-19 23:37:03,115 [INFO] >> cyto3 << model set to be used
2024-12-19 23:37:03,247 [INFO] >>>> loading model C:\Users\Bagnall_Lab\.cellpose\models\cyto3
2024-12-19 23:37:06,428 [INFO] >>>> model diam_mean =  30.000 (ROIs rescaled to this size during training)
{'lowhigh': None, 'percentile': [1.0, 99.0], 'normalize': True, 'norm3D': True, 'sharpen_radius': 0, 'smooth_radius': 0, 'tile_norm_blocksize': 0, 'tile_norm_smooth3D': 1, 'invert': False}
2024-12-19 23:37:06,440 [INFO] channels set to [0, 0]
2024-12-19 23:37:06,441 [INFO] ~~~ FINDING MASKS ~~~
2024-12-19 23:37:06,441 [INFO] multi-stack tiff read in as having 43 planes 1 channels
2024-12-19 23:37:06,580 [INFO] running YX: 43 planes of size (399, 911)
2024-12-19 23:37:06,581 [INFO] 0%|          | 0/43 [00:00<?, ?it/s]
2024-12-19 23:37:22,244 [INFO] 100%|##########| 43/43 [00:15<00:00,  2.75it/s]
2024-12-19 23:37:22,330 [INFO] running ZY: 399 planes of size (43, 911)
2024-12-19 23:37:22,330 [INFO] 0%|          | 0/399 [00:00<?, ?it/s]
2024-12-19 23:37:29,731 [INFO] 100%|##########| 399/399 [00:07<00:00, 53.91it/s]
2024-12-19 23:37:29,804 [INFO] running ZX: 911 planes of size (43, 399)
2024-12-19 23:37:29,804 [INFO] 0%|          | 0/456 [00:00<?, ?it/s]
2024-12-19 23:37:39,267 [INFO] 100%|##########| 456/456 [00:09<00:00, 48.19it/s]
2024-12-19 23:37:39,627 [INFO] network run in 33.05s
NET ERROR: tensors used as indices must be long, byte or bool tensors

**Package version**
cellpose                  3.1.0                    pypi_0    pypi
cudatoolkit               11.3.1               h59b6b97_2
numpy                     1.26.3                   pypi_0    pypi
python                    3.10.16              h4607a30_1
pytorch                   1.12.0          py3.10_cuda11.3_cudnn8_0    pytorch
pyqt6                     6.8.0                    pypi_0    pypi
pyqtgraph                 0.13.7                   pypi_0    pypi
numba                     0.60.0                   pypi_0    pypi

**Comment by simonlsp** [comment]
Same bug here.

**Comment by sun1000yao** [comment]
@simonlsp Thank you! Downgrading to 3.0.8 solved the problem.

**Comment by jpcornil-git** [comment]
FWIW, I don't see a stack trace here/may not be directly related but I faced a similar error (and saw this post).

To fix it, I had to change [dynamics.py @ L796](https://github.com/MouseLand/cellpose/blob/main/cellpose/dynamics.py#L796)
``` python
M1 = M1[tuple(pt)]
```
with
``` python
M1 = M1[tuple(pt.long())]
```

There may be other places like this in the code.

**Comment by carsen-stringer** [comment]
thanks for pointing this out! I think this may be an issue in older pytorch, since this doesn't error in pytorch 2. the issue is that saving it as a long instead of an int takes more memory. could you try upgrading and see if you still get the issue?
